indexer/feature_algo.cpp

```diff
@@ -124,6 +124,50 @@ m2::PointD GetCenter(FeatureType const & f, int scale)
   }
 }
 
-m2::PointD GetCenter(FeatureType const & f) { return GetCenter(f, FeatureType::BEST_GEOMETRY); }
+m2::PointD GetCenter(FeatureType const & f)
+{
+  return GetCenter(f, FeatureType::BEST_GEOMETRY);
+}
+
+double GetMinDistance(FeatureType const & ft, m2::PointD const & pt, int scale)
+{
+  double res = numeric_limits<double>::max();
+  auto updateDistanceFn = [&] (m2::PointD const & p)
+  {
+    double const d = MercatorBounds::DistanceOnEarth(p, pt);
+    if (d < res)
+      res = d;
+  };
+
+  /// @todo Need to check projection here?
+  EGeomType const type = ft.GetFeatureType();
+  switch (type)
+  {
+  case GEOM_POINT:
+    updateDistanceFn(ft.GetCenter());
+    break;
+
+  case GEOM_LINE:
+    ft.ForEachPoint(updateDistanceFn, scale);
+    break;
+
+  default:
+    ASSERT_EQUAL(type, GEOM_AREA, ());
+    ft.ForEachTriangle([&] (m2::PointD const & p1, m2::PointD const & p2, m2::PointD const & p3)
+    {
+      updateDistanceFn(p1);
```

Comment:
How about to check that point belongs to area and return zero in this case? Otherwise it's possible to find 3 projections from the point to the triangle and return the minimal one.